xdsl/dialects/builtin.py

```diff
@@ -1022,16 +1022,30 @@ def from_list(
         else:
             raise TypeError(f"Unsupported element type {data_type}")
 
-    def as_tuple(self) -> tuple[int, ...] | tuple[float, ...]:
+    def iter_values(self) -> Iterator[float] | Iterator[int]:
+        """
+        Iterate over "raw" data. This will not apply the datatype restrictions that the
+        array element type would suggest!
+
+        e.g. given a dense<i8: 99999999, 255, 256>, get_values()
+        would return 1234567, 255, 256 and not 135, 255, 0 (mod 256)
+        """
```

Comment:
should this check not rather apply to the `__init__` of DenseArrayBase rather than the iterator of it?

Reply:
I would feel the creation of an illegal `dense<i8: 99999999, 255, 256>` is the issue here, not the fact i want to iterate over it

Reply:
TBH this will no longer hold after we move to the bytes representation, so I'm not sure that it's worth the effort...

Reply:
On the other hand, we should probably notify users when their input integers are too large for the data type

Reply:
okay, `struct.pack` throws the error if you try to convert a too large value to bytes anyway

Reply:
Ah, good for us to check it first then